core/src/main/java/org/elasticsearch/index/fielddata/IndexFieldData.java

```diff
@@ -140,7 +144,10 @@ public BitSet rootDocs(LeafReaderContext ctx) throws IOException {
              * Get a {@link DocIdSet} that matches the inner documents.
              */
             public DocIdSetIterator innerDocs(LeafReaderContext ctx) throws IOException {
-                Scorer s = innerFilter.scorer(ctx);
+                final IndexReaderContext topLevelCtx = ReaderUtil.getTopLevelContext(ctx);
+                IndexSearcher indexSearcher = new IndexSearcher(topLevelCtx);
+                Weight weight = indexSearcher.createNormalizedWeight(innerQuery, false);
```

Comment:
I took a brief look how this method will be called, but didn't get the complete picture yet. It looks like we might do the weight calculation more often now, not sure how much IndexSearcher creation and creating the weight costs?